src/clients/near.ts

```diff
@@ -614,40 +634,71 @@ export class NearBridgeClient {
   }
 
   // =====================================================================
-  // BITCOIN BRIDGE METHODS (mirrors Rust SDK NearBridgeClient + BtcConnector)
+  // UTXO OPERATIONS (BTC & ZCASH)
   // =====================================================================
 
-  /**
-   * Get Bitcoin deposit address from Satoshi Bridge (BTC -> NEAR flow start)
-   * Mirrors get_btc_address() from Rust SDK
-   */
-  async getBitcoinDepositAddress(
+  private assertUtxoChain(chain: ChainKind): asserts chain is UtxoChain {
+    if (chain !== ChainKind.Btc && chain !== ChainKind.Zcash) {
+      throw new Error(`Unsupported UTXO chain: ${ChainKind[chain] ?? chain}`)
+    }
+  }
+
+  private getUtxoService(chain: UtxoChain): UtxoChainService {
+    const service = this.utxoServices[chain]
+    if (service) {
+      return service
+    }
+
+    if (chain === ChainKind.Zcash) {
+      throw new Error("Zcash support requires initializing NearBridgeClient with a zcashApiKey")
+    }
+
+    throw new Error(`Unsupported UTXO chain: ${ChainKind[chain] ?? chain}`)
+  }
+
+  private getUtxoChainLabel(chain: UtxoChain): string {
+    return UTXO_CHAIN_LABELS[chain] ?? ChainKind[chain] ?? String(chain)
+  }
+
+  private getUtxoConnector(chain: UtxoChain) {
+    switch (chain) {
+      case ChainKind.Btc:
+        return {
+          connector: addresses.btc.btcConnector,
+          token: addresses.btc.btcToken,
+        }
+      case ChainKind.Zcash:
+        this.getUtxoService(chain)
+        return {
+          connector: addresses.zcash.zcashConnector,
+          token: addresses.zcash.zcashToken,
+        }
+      default:
+        throw new Error(`Unsupported UTXO chain: ${ChainKind[chain] ?? chain}`)
+    }
+  }
+
+  async getUtxoDepositAddress(
```

Comment:
Should we add UTXOChainKind type and restrict the input based on that? Then `assertUtxoChain` may not be needed